src/SFA.DAS.AANHub.Api/Controllers/AdminsController.cs

```diff
@@ -28,16 +30,30 @@ public AdminsController(ILogger<AdminsController> logger, IMediator mediator)
         [Produces("application/json")]
         [ProducesResponseType(typeof(string), StatusCodes.Status400BadRequest)]
         [ProducesResponseType(typeof(string), StatusCodes.Status201Created)]
-        public async Task<IActionResult> CreateAdmin([FromHeader(Name = Constants.PostRequestHeaders.RequestedByUserHeader), Required] Guid? userId, CreateAdminModel request)
+        public async Task<IActionResult> CreateAdmin([FromHeader(Name = Constants.PostRequestHeaders.RequestedByUserHeader)] [Required] Guid? userId,
+            CreateAdminModel request)
         {
             _logger.LogInformation("AAN Hub API: Received command to add admin by UserId: {userId}", userId);
 
             var command = (CreateAdminMemberCommand)request;
             command.RequestedByMemberId = userId;
 
             var response = await _mediator.Send(command);
-            return response.IsValidResponse ? new CreatedAtActionResult(nameof(CreateAdmin), "Admins", new { id = response.Result.MemberId }, response.Result)
-                : new BadRequestObjectResult(response.Errors);
+
+            return GetPostResponse(response,
+                new BaseRequestDetails
+                {
+                    ActionName = nameof(CreateAdmin),
+                    ControllerName = "Admins",
+                    GetParameters = response.Errors.Any()
```

Comment:
No need to check anything here in my opinion, we can set this irrespectively like `response.Result?.MemberId.ToString()` this will get ignored if there are any errors.

Reply:
Logic updated as suggested